### PR TITLE
Adds Asset Metrics submodule

### PR DESCRIPTION
**Pull request by rob5standingby**
## Why?
- We need Asset Metrics!

## What changed?
- Added an `AssetMetrics` submodule to perform CRUD operations on asset metrics
- Added tests

### CODE REVIEW

File: src/assets/assetMetrics.js [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+   * Method: DELETE
+   *
+   * @param {string} assetMetricId The UUID formatted ID of the asset metric
+   *
+   * @returns {Promise}
+   * @fulfill {undefined}
+   * @reject {Error}
+   *
+   * @example
+   * contxtSdk.assets.metrics.delete('4f0e51c6-728b-4892-9863-6d002e61204d')
```
Comment (greg5green) [inline src/assets/assetMetrics.js]: Can you add a semi-colon?

File: src/assets/assetMetrics.js [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+   * @fulfill {AssetMetricsFromServer}
+   * @reject {Error}
+   *
+   * @example
+   * contxtSdk.assets.metrics
+   *   .getAll('4f0e51c6-728b-4892-9863-6d002e61204d')
+   *   .then((assetMetrics) => console.log(assetMetrics))
+   *   .catch((err) => console.log(err));
+   */
+  getAll(assetTypeId, paginationOptions) {
```
Comment (greg5green) [inline src/assets/assetMetrics.js]: This doesn't actually get all, so would be better as like `getByAssetTypeId`

File: src/assets/assetMetrics.spec.js [hunk 3]
```diff
@@ -0,0 +1,476 @@
...
+        assetMetric = fixture.build('assetMetric');
+        assetMetrics = new AssetMetrics(baseSdk, baseRequest, expectedHost);
+        assetTypeId = fixture.build('assetType').id;
+      });
+
+      it('throws an error if there is no asset type ID provided', function() {
+        promise = assetMetrics.create(null, assetMetric);
+
+        return expect(promise).to.be.rejectedWith(
+          'An asset type ID is required for creating an asset metric.'
```
Comment (greg5green) [inline src/assets/assetMetrics.spec.js]: Can we switch this to `to create` so it matches the other error message?

File: src/assets/assetMetrics.spec.js [hunk 4]
```diff
@@ -0,0 +1,476 @@
...
+      beforeEach(function() {
+        const assetMetrics = new AssetMetrics(
+          baseSdk,
+          baseRequest,
+          expectedHost
+        );
+
+        promise = assetMetrics.getAll();
+      });
+      it('throws an error when the asset type ID is missing', function() {
```
Comment (greg5green) [inline src/assets/assetMetrics.spec.js]: Can you add a blank line before this `it`

File: src/assets/assetMetrics.spec.js [hunk 5]
```diff
@@ -0,0 +1,476 @@
...
+    context(
+      'when there is missing or malformed required information',
+      function() {
+        let assetMetrics;
+
+        beforeEach(function() {
+          assetMetrics = new AssetMetrics(baseSdk, baseRequest, expectedHost);
+        });
+
+        it('throws an error when there is not provided asset metric ID', function() {
```
Comment (greg5green) [inline src/assets/assetMetrics.spec.js]: not -> no

File: src/assets/assetMetrics.js [hunk 6]
```diff
@@ -0,0 +1,262 @@
...
+
+/**
+ * @typedef {Object} AssetMetric
+ * @property {string} assetTypeId UUID corresponding with the asset type
+ * @property {string} createdAt ISO 8601 Extended Format date/time string
+ * @property {string} description
+ * @property {string} id UUID
+ * @property {string} label
+ * @property {string} organizationId UUID corresponding with the organization
+ * @property {string} timeInterval Options are "hourly", "daily", "weekly", "monthly", "yearly"
```
[the remark this answers is not in the document]
Comment (rob5standingby) [reply src/assets/assetMetrics.js]: It doesn't appear that way. The endpoint requires you to provide one and there's nothing configured in the table for a default value.

File: src/assets/assetMetrics.js (hunk 1 above)
Comment (rob5standingby) [reply src/assets/assetMetrics.js]: yessir!

File: src/assets/assetMetrics.js (hunk 2 above)
Comment (rob5standingby) [reply src/assets/assetMetrics.js]: That sounds good!

File: src/assets/assetMetrics.spec.js (hunk 3 above)
Comment (rob5standingby) [reply src/assets/assetMetrics.spec.js]: 👍 Yeah, that sounds great

File: src/assets/assetMetrics.spec.js (hunk 4 above)
Comment (rob5standingby) [reply src/assets/assetMetrics.spec.js]: I'm on it!

File: src/assets/assetMetrics.spec.js (hunk 5 above)
Comment (rob5standingby) [reply src/assets/assetMetrics.spec.js]: 🤦‍♂️

**Pull request merged**